src/jllvm/object/ClassObject.hpp

```diff
@@ -737,14 +754,25 @@ class ClassObject final : private llvm::TrailingObjects<ClassObject, VTableSlot>
         return offsetof(ClassObject, m_initialized);
     }
 
+    bool markedInitialized() const
+    {
+        return m_initialized == InitializationStatus::UnderInitialization
+               || m_initialized == InitializationStatus::Initialized;
+    }
+
     bool isInitialized() const
     {
-        return m_initialized;
+        return m_initialized == InitializationStatus::Initialized;
+    }
+
+    void markInitialized()
+    {
+        m_initialized = InitializationStatus::UnderInitialization;
     }
 
-    void setInitialized(bool isInitialized)
+    void setInitialized()
     {
-        m_initialized = isInitialized;
+        m_initialized = InitializationStatus::Initialized;
     }
```

Comment:
I like the names of the enum values a lot but think the methods here are a bit confusing. What do you think about having a `setInitializationStatus` method, a `isUnintialized` method and a `isInitialized` method? `UnderInitialization` would return false for either of the two. We can add either a `getInitializationStatus` or `isUnderInitialization` once we need it, but I think the two mentioned should cover the majority of use cases.